lib/src/license_detection/primary_filter.dart

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:pana/src/license_detection/license.dart';
+
+/// Filters [knownLicenses] and returns a list of licenses that could likely be a possible match for input.
+///
+/// A license might be a possible match from the list of [knownLicenses].
+/// This funtction filters from the known licenses by weighing the
+/// number of similar tokens through [tokenSimilarity] method and
+/// returns a list of [License] which have a score above certain threshold.
+List<License> filter(
+        Map<String, int> occurrences, List<License> knownLicenses) =>
```

Comment:
nit: consider using a trailing comman, when the list of argument can't fit on a single line... Some people don't care, and it's not important, just personal preference..
```suggestion
        Map<String, int> occurrences, List<License> knownLicenses,) =>
```